src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -241,6 +332,75 @@ Parameter IStreamsExecutor::Config::GetConfig(const std::string& key) const {
+    const auto num_small_cores =
+        custom::info::default_concurrency(custom::task_arena::constraints{}.set_core_type(core_types.front()));
+    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
+
+    // limit streams to reasonable range
+    int streams = num_streams > base_streams_total ? base_streams_total : std::max(1, num_streams);
+    // if num_threads is set, streams must not be greater than num_threads
+    if (num_threads > 0 && streams > num_threads) {
+        streams = num_threads;
+    }
+    if (streams != num_streams) {
+        OPENVINO_WARN << "Number of streams is updated from " << num_streams << " to " << streams;
+    }
+    return streams;
+}
+
+void IStreamsExecutor::Config::UpdateHybridCustomThreads(Config& config,
+                                                         const int num_big_cores_phys,
+                                                         const int num_small_cores) {
+    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
+    const auto num_cores_phys = num_big_cores_phys + num_small_cores;
+
+    config._small_core_offset = num_big_cores_phys * 2;
+    // if nthreads is set, calculate hybrid aware parameters by nthreads
+    if (config._threads) {
+        // limit threads to reasonable range
+        auto threads = config._threads > num_cores_phys ? num_cores_phys : config._threads;
+        // all threads are placed on big cores
+        if (threads <= num_big_cores_phys) {
+            config._big_core_streams = config._streams;
+            config._threads_per_stream_big = std::max(1, threads / config._streams);
+        } else {
+            // big cores first, then small cores
+            // big_streams = all_streams * big_cores / base_streams
+            config._big_core_streams =
+                config._streams * num_big_cores_phys % base_streams_total == 0
+                    ? config._streams * num_big_cores_phys / base_streams_total
+                    : (config._streams * num_big_cores_phys + base_streams_total) / base_streams_total;
+            config._small_core_streams = config._streams - config._big_core_streams;
+            config._threads_per_stream_big = std::max(1, num_big_cores_phys / config._big_core_streams);
+            if (config._small_core_streams > 0) {
+                // Normally, threads_per_stream_small = threads_per_stream_big * 2, but special conditions are as
+                // follows: Pcore=8, Ecore=4, custom set nthreads=10, nstreams=10, calculate result: big_streams=8,
+                // small_streams=2. There are only 2 threads left, threads_per_stream_small can only be 1.
+                const int remain_threads = threads - config._threads_per_stream_big * config._big_core_streams;
+                config._threads_per_stream_small =
+                    std::min(remain_threads / config._small_core_streams, config._threads_per_stream_big * 2);
```

Comment:
For the case of threads  > num_big_cores_phys:
    Assume:  
            `CPU is 8 PCore + 8 ECore and num_threads = 16, num_streams = 8`
    Per above algorithm we can get the result:
```
            P Core:   6 streams, thread_per_stream = 1
            E Core:   2 streams, thread_per_stream = 2 
```
   It is expected result?